en_US/data-integration/schema-registry-example-protobuf.md

```diff
@@ -0,0 +1,185 @@
+
+### Create the rule
+
+1. In the Dashboard, select **Integration** -> **Rules** from the navigation menu.
+
+2. On the **Rules** page, click **Create** at the top right corner.
+
+3. Use the schema you have just created to write the rule SQL statement:
+
+   ```sql
+   SELECT
+     schema_encode('protobuf_person', json_decode(payload), 'Person') as protobuf_person
+   FROM
+     "protobuf_in"
+   ```
+
+   The key point here is `schema_encode('protobuf_person', json_decode(payload), 'Person')`:
+
+   - The `schema_encode` function encodes the contents of the payload field according to the schema `protobuf_person`;
+   - `as person` stores the encoded value in the variable `person`;
+   - The last argument `Person` specifies that the message type in the payload is the `Person` type defined in the Protobuf schema.
+   - `json_decode(payload)` is needed because `payload` is generally a JSON-encoded binary, and `schema_encode` requires a Map as its input.
+
+4. Click **Add Action**.  Select `Republish` from the drop-down list of the **Action** field. 
+5. In the **Topic** field, type `protobuf_out` as the destination topic.
+6. In the **Payload** field, type message content template: `${protobuf_person}`.
+
+This action sends the Protobuf encoded user message to the topic `protobuf_out`. `${protobuf_person}` is a variable placeholder that will be replaced at runtime with the value of the result of `schema_encode` (a binary value).
+
+### Prepare Device-Side Code
+
+Once the rules have been created, you can simulate the data for testing.
+
+The following code uses the Python language to fill a user message, encode it as binary data, then send it to the `protobuf_in` topic. See [full code](https://gist.github.com/thalesmg/c5f03f99f982401d16ef6583e30144fa) for details.
+
+```python
+def on_message(client, userdata, msg):
+    p = person_pb2.Person()
+    p.ParseFromString(msg.payload)
+    print(msg.topic+" "+str(p))
+```
+
+### Check Rule Execution Results
+
+1) In the Dashboard, select **Diagnose** -> **WebSocket Client**. 
+2) Fill in the connection information for the current EMQX instance.
+   - If you run EMQX locally, you can use the default value.
+   - If you have changed EMQX's default configuration. For example, the configuration change on authentication can require you to type in a username and password.
+
+3. Click **Connect** to connect to the EMQX instance as an MQTT client.
+
+4. In the **Publish** area, type `protobuf_out` in the **Topic** field and type the following message in the **Payload** field:
+
+   ```json
+   {"name":"Shawn","id":1,"email":"shawn@example.com"}
+   ```
+
+5.  Click **Publish**.
+
+6. Install the Python dependencies and execute the device-side code:
+
+   ```shell
+   $ pip3 install protobuf paho-mqtt
+   
+   $ python3 protobuf_mqtt_sub.py
+   Connected with result code 0
+   protobuf_out name: "Shawn"
+   id: 1
+   email: "shawn@example.com"
+   ```
```

Comment:
In this code example, there is no binary message like the one in the Avro example? Just be curious. @thalesmg

Reply:
It is encoded as binary, just the example code was not printing the raw binary payload.  I've updated the example code and docs to include the binary printout.

Reply:
Ok, thank you!